app/[username]/page.tsx

```
@@ -118,8 +121,77 @@ export default async function PublicProfile({
     bgStyle.backgroundImage = "linear-gradient(180deg, #09090b 0%, #1e1b4b 100%)";
   }
 
+  const cookieStore = await cookies();
+  const assignments: { parentId: string; variant: "A" | "B" }[] = [];
+
+  const rawLinks = user.links || [];
+  const abTestGroups = new Map<string, any[]>();
+  const preFilteredLinks: any[] = [];
+
+  const selectVariant = (parentId: string, variants: any[]) => {
```

ESLint

[error] Unexpected any. Specify a different type.

(`@typescript-eslint/no-explicit-any`)

---

[error] Error: Cannot call impure function during render

`Math.random` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

/home/jailuser/git/app/[username]/page.tsx:137:23
  135 |       chosenVariant = cookieVal;
  136 |     } else {
> 137 |       chosenVariant = Math.random() > 0.5 ? "A" : "B";
      |                       ^^^^^^^^^^^^^ Cannot call impure function
  138 |       assignments.push({ parentId, variant: chosenVariant });
  139 |     }
  140 |     return variants.find((v: any) => v.abTestVariant === chosenVariant) || variants[0];

(react-hooks/purity)